leakcanary-android-leak-deobfuscation-plugin/src/main/java/com/squareup/leakcanary/leak/deobfuscation/plugin/LeakcanaryLeakDeobfuscationPlugin.kt

```diff
@@ -0,0 +1,109 @@
+package com.squareup.leakcanary.leak.deobfuscation.plugin
+
+import com.android.build.gradle.AppExtension
+import com.android.build.gradle.LibraryExtension
+import com.android.build.gradle.api.ApplicationVariant
+import com.android.build.gradle.api.BaseVariant
+import com.android.build.gradle.api.LibraryVariant
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.UnknownTaskException
+import org.gradle.api.tasks.TaskProvider
+
+class LeakcanaryLeakDeobfuscationPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    val leakCanaryExtension =
+      project.extensions.create("leakCanary", LeakCanaryDeobfuscationExtension::class.java)
+
+    val androidAppPluginHandler = { _: Plugin<*> ->
```

Comment:
can you do:
```
val extension = project...getByType(BaseExtension::class.java)
val variant = when(extension) { 
 is AppExtension -> appExtension.applicationVariants
 is LibraryExtension -> appExtension.libraryVariants
}
setupPlugin(project, variant, leakCanaryExtension)
```

Reply:
you could also apply the filter before the call to setup and not pass the extension.